client/client_update.go

```diff
@@ -66,6 +66,7 @@ func (u *CurrentUpdate) MarshalJSON() ([]byte, error) {
 	}
 	u.Provides["artifact_name"] = u.Artifact
 	u.Provides["device_type"] = u.DeviceType
+	u.UpdateControlMap = true
```

Comment:
Always set to true for newer clients henceforth. A little too magic this way maybe? :thinking:

Reply:
This is what is stored in the database, correct?

I think it's not necessary to store this information actually, because it is already stored as part of the active maps. And you are not using it anyway, AFAICS..?

Reply:
No, this is what is sent by the client, when it reports it's own current update.
The point being, this way, newer clients will always have the `update_control_map` field  set to `true` in the request.
(see `client_update.go` for usage). 
But like I said, it is probably a little over the top magic wise. Maybe it should be explicitly set in the initialization of the struct instead :thinking:

Reply:
Ah, I completely misread this. Yes, the functionality is correct, but this seems a bit too magical, to do it "behind your back" in `MarshalJSON`. I agree it should be moved to struct initialization.